code/modules/mob/living/carbon/human/species_types/ghost.dm

```diff
@@ -0,0 +1,256 @@
+		BODY_ZONE_R_ARM = /obj/item/bodypart/arm/right/ghost,
+	)
+
+	///Boolean on whether this species type is available at roundstart during halloween, used to deny subtypes.
+	var/halloween_exclusive = TRUE
+
+/datum/species/spirit/check_roundstart_eligible()
+	if(check_holidays(HALLOWEEN) && halloween_exclusive)
+		return TRUE
+	return ..()
+
+/datum/species/spirit/get_physical_attributes()
+	return "Spirits are the spiritual remains of long-passed entities. They lack legs, can fly, but still eat, breathe, hear and see."
+
+/datum/species/spirit/get_species_description()
+	return "Spirits are spirits of long-dead creatures whom, for one reason or another, still roam around."
+
+/datum/species/spirit/get_species_lore()
+	return list(
+		"Spirits are the non-physical remains that linger onto their mortal coil. \
+		They still need their protein and organs to keep themselves \"alive\", \
+		which leads to many of them still believing they are still part of the living, \
+		whether or not they are is a very open-ended debate between philosophers.",
+	)
+
+/datum/species/spirit/create_pref_unique_perks()
+	var/list/to_add = list()
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_POSITIVE_PERK,
+		SPECIES_PERK_ICON = "body",
+		SPECIES_PERK_NAME = "Leg-less",
+		SPECIES_PERK_DESC = "Ghosts lack legs and float, preventing you from falling into holes in the ground.",
+	))
+
+	to_add += list(list(
+		SPECIES_PERK_TYPE = SPECIES_NEGATIVE_PERK,
+		SPECIES_PERK_ICON = "shoe-prints",
+		SPECIES_PERK_NAME = "No Feet",
+		SPECIES_PERK_DESC = "You lack feet, therefore the ability to wear any shoes!",
+	))
+
+	return to_add
+
+/**
+ * Ghost subtype
+ * This is the type of ghost that can actually phase through walls,
+ * exclusive to magic mirrors & admins, as roundstart-ability to phase anywhere
+ * is not something that is generally fun to play against.
+ */
+/datum/species/spirit/ghost
+	name = "Ghost"
+	id = SPECIES_GHOST
+	inherent_traits = list(
+		TRAIT_GENELESS,
+		TRAIT_NEVER_WOUNDED,
+		TRAIT_NOBLOOD,
+		TRAIT_NODISMEMBER,
+		TRAIT_NO_DNA_COPY,
+		TRAIT_NO_PLASMA_TRANSFORM,
+		TRAIT_NO_UNDERWEAR,
+		TRAIT_UNHUSKABLE,
+		TRAIT_NO_FLOATING_ANIM,
+		TRAIT_MOVE_FLYING,
+		//ghost-unique
+		TRAIT_SEE_BLESSED_TILES,
+	)
+	//they have a different chest.
+	bodypart_overrides = list(
+		BODY_ZONE_HEAD = /obj/item/bodypart/head/ghost,
+		BODY_ZONE_CHEST = /obj/item/bodypart/chest/ghost,
+		BODY_ZONE_L_ARM = /obj/item/bodypart/arm/left/ghost,
+		BODY_ZONE_R_ARM = /obj/item/bodypart/arm/right/ghost,
+	)
+	changesource_flags = MIRROR_BADMIN | MIRROR_PRIDE | MIRROR_MAGIC
+	halloween_exclusive = FALSE
+
+	///Innate passthrough ability given to ghosts that allows them to phase but drops their stuff.
+	var/datum/action/innate/toggle_passthrough/passthrough_ability
+
+/datum/species/spirit/ghost/get_physical_attributes()
+	return "Ghosts are the spiritual remains of long-passed entities. They lack legs, can fly, can choose at will to become opaque, \
```

Comment:
```suggestion
	return "Ghosts are the spiritual remains of long-passed entities. They lack legs, can fly, can choose at will to become incorporeal, \
```
Opaque just refers to becoming non-transparent, unless that's what you're going for for the lore here